edb/server/protocol/auth_ext/oauth.py

```diff
@@ -30,44 +30,39 @@ class Client:
     provider: base.BaseProvider
 
     def __init__(
-        self,
-        db: Any,
-        provider_name: str,
-        base_url: str | None = None
+        self, db: Any, provider_name: str, base_url: str | None = None
     ):
         self.db = db
 
         http_factory = lambda *args, **kwargs: http_client.HttpClient(
             *args, edgedb_test_url=base_url, **kwargs
         )
 
+        provider_args = self._get_provider_config(provider_name)
+        provider_kwargs = {
+            "http_factory": http_factory,
+            "additional_scope": self._get_additional_scope(provider_name),
+        }
+
         match provider_name:
             case "builtin::oauth_github":
-                self.provider = github.GitHubProvider(
-                    *self._get_provider_config(provider_name),
-                    http_factory=http_factory,
-                )
+                provider_class = github.GitHubProvider  # type: ignore
```

Comment:
I couldn't get `mypy` to shut-up about the type of `provider_class`, even if I annotated the variable like `provider_class: Type[base.BaseProvider]`, so I just silenced it.

This refactor seemed worth it since it feels error prone to manually have every branch of this match have to instantiate the provider with the exact same arguments. Especially an "optional" argument like this, it would be easy to forget to add it in one branch.

Reply:
That's OK, but what was mypy saying?

Reply:
```
edb/server/protocol/auth_ext/oauth.py:51:34: error: Incompatible types in assignment (expression has type "Type[GoogleProvider]", variable has type "Type[GitHubProvider]")  [assignment]
edb/server/protocol/auth_ext/oauth.py:53:34: error: Incompatible types in assignment (expression has type "Type[AzureProvider]", variable has type "Type[GitHubProvider]")  [assignment]
edb/server/protocol/auth_ext/oauth.py:55:34: error: Incompatible types in assignment (expression has type "Type[AppleProvider]", variable has type "Type[GitHubProvider]")  [assignment]
```

Reply:
Weird. @msullivan can you teach us to use mypy properly here? I'm mostly curious at this point, it's not urgent

Reply:
(we can merge as is obviously)